libs/orchestrator-ui-components/src/lib/components/SettingsPage/EngineStatusButton.tsx

```diff
@@ -0,0 +1,46 @@
+import { EuiButton } from '@elastic/eui';
+import React from 'react';
+
+interface EngineStatusButtonProps {
+    isLoading: boolean;
+    isRunning: boolean;
+    changeEngineStatus: () => void;
+}
+
+export const EngineStatusButton = ({
+    isLoading,
+    isRunning,
+    changeEngineStatus,
+}: EngineStatusButtonProps) => {
+    if (isLoading) {
+        return (
+            <EuiButton isLoading fill>
+                Loading...
+            </EuiButton>
+        );
+    } else {
```

Comment:
Multiple levels of if-else might be confusing to read. Suggesting to remove the whole else block and rewrite it like this:

`return isRunning ? <EuiButtonFromLine24> : <EuiButtonFromLine35>`

Reply:
I will fix this